src/sortStudents.ts

```diff
@@ -1,16 +1,64 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function calculateAverageGrade(grades: number[]): number {
+  return grades.reduce((prev, grade) => prev + grade, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copiedStudents: Student[] = [...students];
+
+  copiedStudents.sort((student1, student2): number => {
+    let result: number = 0;
+
+    switch (sortBy) {
+      case SortType.Name:
+        result = student1.name.localeCompare(student2.name);
+        break;
+
+      case SortType.Surname:
+        result = student1.surname.localeCompare(student2.surname);
+        break;
```

Comment:
Check the checklist. You can combine these. Also the same applies to Age and Married
```suggestion
      case SortType.Name:
      case SortType.Surname:
        result = student1[sortBy].localeCompare(student2[sortBy]);
        break;
```